samples/browser/src/main/java/org/mozilla/samples/browser/addons/AddonsFragment.kt

```diff
@@ -188,25 +227,41 @@ class AddonsFragment : Fragment(), View.OnClickListener {
         }
 
         private fun createListWithSections(addons: List<Addon>): List<Any> {
-            // We want to have the installed add-ons first in the list.
-            val sortedAddons = addons.sortedBy { !it.isInstalled() }
-
             val itemsWithSections = ArrayList<Any>()
-            val shouldAddInstalledSection = sortedAddons.first().isInstalled()
-            var isRecommendedSectionAdded = false
+            val installedAddons = ArrayList<Addon>()
+            val recommendedAddons = ArrayList<Addon>()
 
-            if (shouldAddInstalledSection) {
+            addons.forEach { addon ->
+                if (addon.isInstalled()) {
+                    if (addon.isNotSupported()) {
+                        unsupportedAddons.add(addon)
+                    } else {
+                        installedAddons.add(addon)
+                    }
+                } else {
+                    recommendedAddons.add(addon)
+                }
+            }
+
+            // Add installed section and addons if available
+            if (installedAddons.isNotEmpty()) {
                 itemsWithSections.add(Section(R.string.mozac_feature_addons_installed_section))
+
+                itemsWithSections.addAll(installedAddons)
             }
 
-            sortedAddons.forEach { addon ->
-                if (!isRecommendedSectionAdded && !addon.isInstalled()) {
-                    itemsWithSections.add(Section(R.string.mozac_feature_addons_recommended_section))
-                    isRecommendedSectionAdded = true
-                }
+            // Add recommended section and addons if available
+            if (recommendedAddons.isNotEmpty()) {
+                itemsWithSections.add(Section(R.string.mozac_feature_addons_recommended_section))
+
+                itemsWithSections.addAll(recommendedAddons)
+            }
 
-                itemsWithSections.add(addon)
+            // Add unsupported section
+            if (unsupportedAddons.isNotEmpty()) {
+                itemsWithSections.add(NotYetSupportedSection(R.string.mozac_feature_addons_unsupported_section))
```

Comment:
I'm bit confused. Do we want to unsupported add-ons items and section label look different from what we already had?

Reply:
Is it possible to reuse the `SectionViewHolder`?

Reply:
I'm not quite sure about the UI, but I was thinking that we will have a list like below, could you help me to clarify?:

**Installed**
     Addon
    Addon
**Recommended**
    Addon
    Addon
**Not support yet**
    Addon
    Addon

Reply:
But What I think we are doing now is:

**Installed**
     Addon
    Addon
**Recommended**
    Addon
    Addon
**Not supported yet** (If you clicked will take you to another screen where will show the unsupported add-ons)

Reply:
@Amejia481 this is easy to miss in the mocks (var 02): https://miro.com/app/board/o9J_kw8Lt8g=/?moveToWidget=3074457347102980262

Reply:
So, this is as desired.

Reply:
@Amejia481 They are different in the mocks

Reply:
Oh, thanks. I missed on the mockups 👍